### ISSUE TITLE
is max_seq_len configurable or hardcoded parameter?

### ISSUE THREAD

**Issue by morphpiece** [karpathy/llm.c#569]
Today I was going to train a gpt3_124m model, when I noticed that the max_seq_len is hardcoded [here](https://github.com/karpathy/llm.c/blob/d396cd18b71367f79cbaab8f8203e64e578f9ee8/train_gpt2.cu#L653) and at the same it's a configurable parameter [here](https://github.com/karpathy/llm.c/blob/d396cd18b71367f79cbaab8f8203e64e578f9ee8/train_gpt2.cu#L1513). Then I executed the training executable with and without `-t 2048` and find that the number of model parameters stays the same at 124475904. 

is there a 🐛 bounty? 😃

**Comment by gordicaleksa** [comment]
yeah, we're focused on using the exact setup of gpt-2/3 right now, the idea is not to make this as configurable as possible, many things are hardcoded which goes against the general SWE principles, but it's less important given the goal of this project.

**Comment by morphpiece** [comment]
Two points come to mind. 

It's fine to hardcode at this stage (like vocabulary), but then it should **not** be available as a configurable parameter. That can throw someone off completely. It's better to fix it like vocabulary is fixed and not be available as a configurable input.

Secondly, since one of the usecases is GPT-3 (which uses max_seq_len of 2048), it is assumed that users will change this value from the hardcoded 1024 to 2048. In that respect, probably hardcoding this param needs to be reconsidered. 

Thanks again for this amazing work!
